crates/biome_js_analyze/src/lint/style/use_naming_convention.rs

```diff
@@ -1844,18 +1846,27 @@ pub enum Scope {
     Global,
 }
 
+declare_node_union! {
+    pub AnyJsGlobalScope =
+        JsModule
+        | JsScript
+        | TsDeclarationModule
+        | TsModuleDeclaration
+        | TsExternalModuleDeclaration
+}
+
 impl Scope {
     /// Returns the scope of `node` or `None` if the scope cannot be determined or
     /// if the scope is an external module.
     fn from_declaration(node: &AnyJsBindingDeclaration) -> Option<Scope> {
-        let control_flow_root = node
-            .syntax()
-            .ancestors()
-            .skip(1)
-            .find(|x| AnyJsControlFlowRoot::can_cast(x.kind()))?;
+        let control_flow_root = node.syntax().ancestors().skip(1).find(|x| {
+            AnyJsControlFlowRoot::can_cast(x.kind())
```

Comment:
I didn't add `TsDeclarationModule` in `AnyJsControlFlowRoot` because a declaration file has no control flow.